feat(tokens): DLT-1696 update docsite for new themes

braddialpad opened the pull request:
# feat(tokens): DLT-1696 update docsite for new themes

<!--- Feel free to remove any unused sections -->

## Obligatory GIF (super important!)

<!--- go to giphy.com, pick a gif, click share, copy gif link, paste within the () brackets below. -->

![Obligatory GIF](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExeTljaTdncGlzbGZ6Z2tndDk2OTQ2M3V0ZW8xMGppb2J6MXhzNGNwciZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/NbKVs8U9udQRCXucMh/giphy-downsized-large.gif)

## :hammer_and_wrench: Type Of Change

<!--- Tick or place an `x` for the type of change. Should match the type in the commit message / PR title
in https://github.com/dialpad/dialtone/blob/staging/.github/COMMIT_CONVENTION.md -->

- [x] Feature

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-1696

## :book: Description

NOTE: this PR is split into 3 parts for ease of review. Token processing, PostCSS processing, and Documentation site updates. This is the documentation site updates which includes updating the doc site to handle the new tokens system.

- Added brand selector on the tokens page. This will allow you to display the tokens for a specific brand (dialpad / tmo / expressive) in addition to just a theme (dark / light).
- Use the new isCompositionToken property to prevent displaying "sub-composition" tokens.
- Added a new theme function in dialtone/common that allows you to set the theme and brand. Implemented this into the "theme changer" on the dialtone docsite. This will also set data-dt-brand and data-dt-theme on the root element for ease of use on the application side.

## :bulb: Context

Major changes in theme tokens, docsite needs be updated to document and display them.

## :pencil: Checklist

<!--- Tick or place an `x` in all of the checkboxes that apply -->
<!--- Remove checkboxes that do not apply -->

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

## :camera: Screenshots / GIFs

![Screenshot 2024-07-24 at 3 15 10 PM](https://github.com/user-attachments/assets/58e4540d-19b6-4274-8579-a2a87feb538f)

juliodialpad committed:
fix(emoji-text-wrapper): DLT-1699 emoji alignment (#407)

bianca-artola-dialpad committed:
fix(multi-select): NO-JIRA update input padding when resizing (#413)

ninamarina committed:
fix: DLT-1899 page TOC persisting from page to page (#414)

hynes-dialpad committed:
docs(writing-guidelines): DDT-554 ux content writing guidelines v2 (#352)

Co-authored-by: Julio Ortega <julio.ortega@dialpad.com>
Co-authored-by: Francis Rupert <francis.rupert@dialpad.com>

semantic-release-bot committed:
chore(release): NO-JIRA dialtone/v9.56.2

## [9.56.2](https://github.com/dialpad/dialtone/compare/dialtone/v9.56.1...dialtone/v9.56.2) (2024-07-24)

### Bug Fixes

* DLT-1899 page TOC persisting from page to page ([#414](https://github.com/dialpad/dialtone/issues/414)) ([86759dc](https://github.com/dialpad/dialtone/commit/86759dca3727d52b723edf62d6d6c0f22bd432a2))
* **Emoji Text Wrapper:** DLT-1699 emoji alignment ([#407](https://github.com/dialpad/dialtone/issues/407)) ([bac9692](https://github.com/dialpad/dialtone/commit/bac9692091f32763c3b4227a46d3c94c0b9091a9))
* **Multi Select:** NO-JIRA update input padding when resizing ([#413](https://github.com/dialpad/dialtone/issues/413)) ([06bc49d](https://github.com/dialpad/dialtone/commit/06bc49dd78d8f8c168b5b0af024da0ecb3bd5c51))

### Documentation

* **Writing Guidelines:** DDT-554 ux content writing guidelines v2 ([#352](https://github.com/dialpad/dialtone/issues/352)) ([d9819a0](https://github.com/dialpad/dialtone/commit/d9819a0c75f4ca1c936bb4dfc5bf845000631412))

semantic-release-bot committed:
chore(release): NO-JIRA dialtone-css/v8.38.2

## [8.38.2](https://github.com/dialpad/dialtone/compare/dialtone-css/v8.38.1...dialtone-css/v8.38.2) (2024-07-24)

### Bug Fixes

* **Emoji Text Wrapper:** DLT-1699 emoji alignment ([#407](https://github.com/dialpad/dialtone/issues/407)) ([bac9692](https://github.com/dialpad/dialtone/commit/bac9692091f32763c3b4227a46d3c94c0b9091a9))

semantic-release-bot committed:
chore(release): NO-JIRA dialtone-vue2/v2.144.2

## [2.144.2](https://github.com/dialpad/dialtone/compare/dialtone-vue2/v2.144.1...dialtone-vue2/v2.144.2) (2024-07-24)

### Bug Fixes

* **Emoji Text Wrapper:** DLT-1699 emoji alignment ([#407](https://github.com/dialpad/dialtone/issues/407)) ([bac9692](https://github.com/dialpad/dialtone/commit/bac9692091f32763c3b4227a46d3c94c0b9091a9))
* **Multi Select:** NO-JIRA update input padding when resizing ([#413](https://github.com/dialpad/dialtone/issues/413)) ([06bc49d](https://github.com/dialpad/dialtone/commit/06bc49dd78d8f8c168b5b0af024da0ecb3bd5c51))

semantic-release-bot committed:
chore(release): NO-JIRA dialtone-vue3/v3.137.2

## [3.137.2](https://github.com/dialpad/dialtone/compare/dialtone-vue3/v3.137.1...dialtone-vue3/v3.137.2) (2024-07-24)

### Bug Fixes

* **Emoji Text Wrapper:** DLT-1699 emoji alignment ([#407](https://github.com/dialpad/dialtone/issues/407)) ([bac9692](https://github.com/dialpad/dialtone/commit/bac9692091f32763c3b4227a46d3c94c0b9091a9))
* **Multi Select:** NO-JIRA update input padding when resizing ([#413](https://github.com/dialpad/dialtone/issues/413)) ([06bc49d](https://github.com/dialpad/dialtone/commit/06bc49dd78d8f8c168b5b0af024da0ecb3bd5c51))

braddialpad committed:
DLT-1696: update docsite for new themes

braddialpad committed:
attach setTheme to window

ninamarina commented:
I am seeing some duplicated tokens in the list:

<img width="986" alt="image" src="https://github.com/user-attachments/assets/1ccae233-0266-4826-9453-6f0b301ca6b3">

This could be my local build though

braddialpad committed:
Merge branch 'theme-tokens-postcss' into theme-tokens-docsite

braddialpad committed:
add globalThis instead of window

braddialpad commented:
> I am seeing some duplicated tokens in the list:
> 
> This could be my local build though

Nope, that appears to be a legitimate problem. Looking into it.

juliodialpad committed:
feat(icons): DLT-1697 add types (#415)

semantic-release-bot committed:
chore(release): NO-JIRA dialtone/v9.57.0

# [9.57.0](https://github.com/dialpad/dialtone/compare/dialtone/v9.56.2...dialtone/v9.57.0) (2024-07-25)

### Features

* **Icons:** DLT-1697 add types ([#415](https://github.com/dialpad/dialtone/issues/415)) ([9adcd81](https://github.com/dialpad/dialtone/commit/9adcd8164de8e25c6b986e7850983088bf584f87))

semantic-release-bot committed:
chore(release): NO-JIRA dialtone-icons/v4.25.0

# [4.25.0](https://github.com/dialpad/dialtone/compare/dialtone-icons/v4.24.0...dialtone-icons/v4.25.0) (2024-07-25)

### Features

* **Icons:** DLT-1697 add types ([#415](https://github.com/dialpad/dialtone/issues/415)) ([9adcd81](https://github.com/dialpad/dialtone/commit/9adcd8164de8e25c6b986e7850983088bf584f87))

ninamarina committed:
docs: DLT-1897 move storybook to doc site - Emoji to Emoji Text Wrapper (#412)

ninamarina commented:
I am also not seeing the examples of the shadows, seems like some of the tokens that make the composition are undefined: 
<img width="411" alt="image" src="https://github.com/user-attachments/assets/30750b44-2d20-4aed-9d11-cb7456861cd7">

braddialpad commented:
> I am also not seeing the examples of the shadows, seems like some of the tokens that make the composition are undefined:

ah yeah looks like only on the focus shadows. thanks

braddialpad committed:
Merge branch 'theme-tokens-postcss' into theme-tokens-docsite

braddialpad commented:
> I am seeing some duplicated tokens in the list:

Silly logic error by me in the doc generation, should be fixed now.

ninamarina commented on apps/dialtone-documentation/docs/.vuepress/baseComponents/tokens/AllTokens.vue:
```diff
@@ -111,7 +113,6 @@ const updateHeadersRecursively = (node, category) => {
 
 onBeforeMount(() => {
   addTokensToStructure(processedTokens);
-  addComposedTokens(processedTokens);
```
You can remove this function definition if it's not being used anywhere anymore

braddialpad committed:
Merge branch 'theme-tokens-postcss' into theme-tokens-docsite

braddialpad commented:
> ah yeah looks like only on the focus shadows. thanks

shadow issue fixed.

ninamarina commented:
The shadows are looking good!

I see that the tooltips for the composition tokens are no longer there:
<img width="273" alt="image" src="https://github.com/user-attachments/assets/924cb0ec-d233-4b2b-82ca-bd46f300ea33">

These are not showing the preview for this PR

braddialpad commented:
> I see that the tooltips for the composition tokens are no longer there:

Yep, workin on that one

francisrupert commented:
When switching to `Expressive` Brand, I'm seeing duplicates of at least font size. Haven't found other duplicates.

https://github.com/user-attachments/assets/1640a92c-752b-4fbb-93fa-bb1e5ed48650

braddialpad committed:
fix composition token tooltips

braddialpad committed:
remove unused

braddialpad commented:
> I see that the tooltips for the composition tokens are no longer there:

These are now fixed!

braddialpad commented:
> When switching to `Expressive` Brand, I'm seeing duplicates of at least font size. Haven't found other duplicates.

Hmm bit of an edge case here.. Seems like in the expressive brand we are redefining base font sizes in a semantic file which is not really supposed to happen. Only semantic tokens are meant to be defined in a semantic file. I'm guessing this is a case we'll need for the expressive theme though right? In the case of a conflict like this between base and semantic I guess we could just have the one from the semantic file override the base. It already works like that in css (just tested it)

braddialpad committed:
merge theme / brand tokens properly

braddialpad commented:
> When switching to `Expressive` Brand, I'm seeing duplicates of at least font size. Haven't found other duplicates.

Fixed! was actually pretty simple

ninamarina commented on apps/dialtone-documentation/docs/.vuepress/baseComponents/tokens/TokenValue.vue:
```diff
@@ -2,16 +2,7 @@
-          </span>
-        </span>
-        <span v-else v-dt-tooltip="getTooltipValue(value)" class="h:d-fc-secondary">
+        <span v-dt-tooltip="getTooltipValue(value)" :class="{ 'h:d-fc-secondary': getTooltipValue(value) }">
```
Why add this condition to `h:d-fc-secondary`? Is there a case where `getTooltipValue(value)` is undefined?

braddialpad replied:
Yeah, it's only for the /

ninamarina reviewed (approved):
Looks really good!

bianca-artola-dialpad committed:
feat(message-input): DLT-1912 add slot for split button (#419)

ninamarina commented:
> > When switching to `Expressive` Brand, I'm seeing duplicates of at least font size. Haven't found other duplicates.
> 
> Hmm bit of an edge case here.. Seems like in the expressive brand we are redefining base font sizes in a semantic file which is not really supposed to happen. Only semantic tokens are meant to be defined in a semantic file. I'm guessing this is a case we'll need for the expressive theme though right? In the case of a conflict like this between base and semantic I guess we could just have the one from the semantic file override the base. It already works like that in css (just tested it)

Should this be fixed in Figma at some point in the future? because it means that some tokens are repeated in the base and in the brand file

braddialpad commented:
> Should this be fixed in Figma at some point in the future? because it means that some tokens are repeated in the base and in the brand file

Yeah, but I guess I'm not entirely sure that's wrong? If you needed to change a base token at theme level like in this case, how else would you do it?

ninamarina commented:
> > Should this be fixed in Figma at some point in the future? because it means that some tokens are repeated in the base and in the brand file
> 
> Yeah, but I guess I'm not entirely sure that's wrong? If you needed to change a base token at theme level like in this case, how else would you do it?

I guess for now is fine, but I understood that in the future we want base tokens to not change with the theme / brand.

braddialpad committed:
Merge branch 'theme-tokens-postcss' into theme-tokens-docsite

braddialpad committed:
output common

braddialpad committed:
Update README.md

braddialpad committed:
update readme documentation

semantic-release-bot committed:
chore(release): NO-JIRA dialtone/v9.58.0

# [9.58.0](https://github.com/dialpad/dialtone/compare/dialtone/v9.57.0...dialtone/v9.58.0) (2024-07-30)

### Documentation

* DLT-1897 move storybook to doc site - Emoji to Emoji Text Wrapper ([#412](https://github.com/dialpad/dialtone/issues/412)) ([dedb229](https://github.com/dialpad/dialtone/commit/dedb229c6e5ebbea8638c0537a8a31c37bf7ef3c))

### Features

* **Message Input:** DLT-1912 add slot for split button ([#419](https://github.com/dialpad/dialtone/issues/419)) ([e50139e](https://github.com/dialpad/dialtone/commit/e50139e659988b70aa349bf492213d81947e5538))

semantic-release-bot committed:
chore(release): NO-JIRA dialtone-vue2/v2.145.0

# [2.145.0](https://github.com/dialpad/dialtone/compare/dialtone-vue2/v2.144.2...dialtone-vue2/v2.145.0) (2024-07-30)

### Documentation

* DLT-1897 move storybook to doc site - Emoji to Emoji Text Wrapper ([#412](https://github.com/dialpad/dialtone/issues/412)) ([dedb229](https://github.com/dialpad/dialtone/commit/dedb229c6e5ebbea8638c0537a8a31c37bf7ef3c))

### Features

* **Message Input:** DLT-1912 add slot for split button ([#419](https://github.com/dialpad/dialtone/issues/419)) ([e50139e](https://github.com/dialpad/dialtone/commit/e50139e659988b70aa349bf492213d81947e5538))

semantic-release-bot committed:
chore(release): NO-JIRA dialtone-vue3/v3.138.0

# [3.138.0](https://github.com/dialpad/dialtone/compare/dialtone-vue3/v3.137.2...dialtone-vue3/v3.138.0) (2024-07-30)

### Documentation

* DLT-1897 move storybook to doc site - Emoji to Emoji Text Wrapper ([#412](https://github.com/dialpad/dialtone/issues/412)) ([dedb229](https://github.com/dialpad/dialtone/commit/dedb229c6e5ebbea8638c0537a8a31c37bf7ef3c))

### Features

* **Message Input:** DLT-1912 add slot for split button ([#419](https://github.com/dialpad/dialtone/issues/419)) ([e50139e](https://github.com/dialpad/dialtone/commit/e50139e659988b70aa349bf492213d81947e5538))

hynes-dialpad committed:
docs: DLT-1918 update do / don't usage well styles (#421)

Co-authored-by: Francis Rupert <francis.rupert@dialpad.com>

braddialpad committed:
update theme apply

braddialpad committed:
Merge branch 'staging' into theme-tokens-docsite

braddialpad committed:
add dialtone-default-theme

braddialpad committed:
update readme, default theme, chunks folder

braddialpad committed:
update to style dictionary 4, fix mobile tokens issues

braddialpad committed:
add blog post

braddialpad committed:
formatting
